src/Workload.cpp

```diff
@@ -254,3 +157,50 @@ JobSpecification Workload::sampleJob(size_t job_id, const size_t infiles_per_job
 
     return job_specification;
 }
+
+template <class InputIt, class OutputIt, class Pred, class Fct>
+void transform_if(InputIt first, InputIt last, OutputIt dest, Pred pred, Fct transform)
+{
+   while (first != last) {
+      if (pred(*first))
+         *dest++ = transform(*first);
+
+      ++first;
+   }
+}
+
+void Workload::assignFiles(std::vector<Dataset> const &dataset_specs)
```

Comment:
Would you mind adding a docstring to this function?

Reply:
Explain what it does and where it is used.

Reply:
So, basically something like:
"Splits the files of matching datasets evenly across all jobs in the workload."